### PR TITLE
refactor: use kafkamgmt API client from SDK

### PR DESCRIPTION
**Pull request by craicoverflow**
This PR updates the CLI to use the latest kafkamgmt client: https://github.com/redhat-developer/app-services-sdk-go/releases/tag/kafkamgmt%2Fv0.1.0

- 5603b83da55660472cfff18ea7c6937a1ed0dd75 adds an OAuth transport layer so that tokens can be refreshed automatically. This means users are less likely to experience an expirt of their token between first instantiate of the API client and when executing a call.

### Verification Steps

Everything should run as usual.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [x] Refactor

### CODE REVIEW

**Comment by wtrocki** [comment]
Rebase needed.

File: Makefile [hunk 1]
```diff
@@ -123,7 +123,7 @@ openapi/kas/validate:
 
 # generate the openapi schema
 openapi/kas/generate:
-	openapi-generator-cli generate -i openapi/kafka-service.yaml -g go --package-name kasclient -p="generateInterfaces=true" --ignore-file-override=$$(pwd)/.openapi-generator-ignore -o ${kasapi_dir}
+	openapi-generator-cli generate -i openapi/kafka-service.yaml -g go --package-name kafkamgmtv1 -p="generateInterfaces=true" --ignore-file-override=$$(pwd)/.openapi-generator-ignore -o ${kasapi_dir}
```
[the remark this answers is not in the document]
Comment (craicoverflow) [reply Makefile:126]: Mocks still need the OpenAPI file. I'm going to keep this open as a running branch to verify integration.

File: pkg/connection/keycloak_connection.go [hunk 2]
```diff
@@ -311,11 +310,25 @@ func (c *KeycloakConnection) createAmsAPIClient() *amsclient.APIClient {
...
-	cfg.AddDefaultHeader("Authorization", fmt.Sprintf("Bearer %v", c.Token.AccessToken))
+	cfg.HTTPClient = c.createOAuthTransport(c.Token.AccessToken)
 
 	apiClient := amsclient.NewAPIClient(cfg)
 
 	return apiClient
 }
+
+// wraps the HTTP client with an OAuth2 Transport layer to provide automatic token refreshing
+func (c *KeycloakConnection) createOAuthTransport(accessToken string) *http.Client {
```
Comment (wtrocki) [inline pkg/connection/keycloak_connection.go:321]: ❤️

**Comment by wtrocki** [comment]
@craicoverflow 

## Review

- [x] Reviewed changes 
- [x] Checked login, config and security (on top of SDK)

## Verification

- [x] Compilation
- [x] Basic usage 

Screenshot 2021-05-24 at 14 44 12

## Comments

I would recommend to move kafka-service.yaml to mock repository. 
I do not see this file being used in any other place currently.

**Review by wtrocki — APPROVED** [review]

**Pull request merged**